src/winrm_client.py

```diff
@@ -0,0 +1,101 @@
+import base64
+import logging
+import os
+import winrm
+from src.base_client import BaseShellClient
+from scripts.powershell_scripts import (
+    create_directory_script,
+    upload_file_script,
+    file_exists_check_script,
+    get_file_checksum_script
+)
+
+class WinRMClient(BaseShellClient):
+    def __init__(self, host, username, password):
+        super().__init__(host, username, password)
+        self.session = winrm.Session(host, auth=(username, password), transport='basic', server_cert_validation='ignore')
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        logging.info(f"Disconnected from {self.host}.")
+        pass
+
+    @staticmethod
+    def read_and_encode_file(local_path):
```

Comment:
It's utility function: can be used for any local file independently on OS type. Please move to utils/file_utils.py